src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2098,10 +2098,11 @@ def aks_scale(cmd, client, resource_group_name, name, node_count, nodepool_name=
         raise CLIError('There are more than one node pool in the cluster. '
                        'Please specify nodepool name or use az aks nodepool command to scale node pool')
 
-    if node_count == 0:
```

Comment:
I wonder if we already validate node_count >= 0 somewhere so you can't pass negative numbers.

Reply:
AgentPool validation will limit the count between 0 to 100, negative number won't be allowed. :)